managed/services/telemetry/config.go

```diff
@@ -183,9 +189,41 @@ func (c *ServiceConfig) Init(l *logrus.Entry) error { //nolint:gocognit
 		}
 	}
 
+	testClickhouseDSN := c.testClickhouseDSN()
+	if testClickhouseDSN != "" {
+		c.l.Debugf("Overriding Telemetry.DataSources.QanDBSelect.DSN with environment variables [%s][%s][%s][%s] to %q.", envClickhouseAddr, envClickHouseDatabase, envClickhouseBlockSize, envClickhousePoolSize, testClickhouseDSN)
+		c.DataSources.QanDBSelect.DSN = testClickhouseDSN
+	}
+
 	return nil
 }
 
+func (c *ServiceConfig) testClickhouseDSN() string {
```

Comment:
I think we should move this logic into `main.go` and get the environment variable via kingpin mechanisms. We are going to implement the same for other data sources.